src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/QualityGuidelines/AvoidMultipleEnumerations/AvoidMultipleEnumerations.cs

```diff
@@ -0,0 +1,333 @@
+            nameof(Enumerable.ThenByDescending),
+            nameof(Enumerable.Where),
+            nameof(Enumerable.Concat),
+            nameof(Enumerable.Except),
+            nameof(Enumerable.GroupJoin),
+            nameof(Enumerable.Intersect),
+            nameof(Enumerable.Join),
+            nameof(Enumerable.Union),
+            nameof(Enumerable.Zip),
+            nameof(Enumerable.DefaultIfEmpty),
+            // Only available on .net6 or later
+            "Chunk",
+            "DistinctBy",
+            "ExceptBy",
+            "IntersectBy",
+            "UnionBy",
+            // Only available on .netstandard 2.1 or later
+            "TakeLast",
+            "SkipLast");
+
+        /// <summary>
+        /// Special Linq methods that no effect on its parameter, and not return new IEnumerable instance.
+        /// </summary>
+        private static readonly ImmutableArray<string> s_noEffectLinqChainMethods = ImmutableArray.Create(
+            nameof(Enumerable.AsEnumerable));
+
+        /// <summary>
+        /// Linq methods don't enumerate deferred type, and is not a linq chain.
+        /// </summary>
+        private static readonly ImmutableArray<string> s_noEnumerationLinqMethods = ImmutableArray.Create(
+            // Only available on .net6 or later
+            "TryGetNonEnumeratedCount");
+
+        protected abstract GlobalFlowStateDictionaryFlowOperationVisitor CreateOperationVisitor(
+            GlobalFlowStateDictionaryAnalysisContext context,
+            WellKnownSymbolsInfo wellKnownSymbolsInfo);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            context.RegisterCompilationStartAction(CompilationStartAction);
+        }
+
+        private void CompilationStartAction(CompilationStartAnalysisContext context)
+            => context.RegisterOperationBlockStartAction(OnOperationBlockStart);
+
+        private void OnOperationBlockStart(OperationBlockStartAnalysisContext operationBlockStartAnalysisContext)
+        {
+            var wellKnownTypeProvider = WellKnownTypeProvider.GetOrCreate(operationBlockStartAnalysisContext.Compilation);
+            var linqChainMethods = GetLinqMethods(wellKnownTypeProvider, s_linqChainMethods);
+            var noEnumerationMethods = GetLinqMethods(wellKnownTypeProvider, s_noEnumerationLinqMethods);
+            var enumeratedMethods = GetEnumeratedMethods(wellKnownTypeProvider, s_immutableCollectionsTypeNamesAndConvensionMethods, s_enumeratedParametersLinqMethods, s_contructorsEnumeratedParameterTypes);
+            var noEffectLinqChainMethods = GetLinqMethods(wellKnownTypeProvider, s_noEffectLinqChainMethods);
+            var compilation = operationBlockStartAnalysisContext.Compilation;
+            var additionalDeferTypes = GetTypes(compilation, s_additionalDeferredTypes);
```

Comment:
HM not sure why these still created here, did you merged your other PR updating these?